run/markdown-preview/editor/test/render.test.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const assert = require('assert');
+const path = require('path');
+const sinon = require('sinon');
+
+let request, service, markdown;
+
+describe('Editor renderRequest unit tests', function () {
+  this.timeout(9000);
+  
+  before(async () => {
+    request = require(path.join(__dirname, '..', 'render'));
+    service = {url: 'https://www.example.com'};
+    markdown = "**markdown text**";
+  });
+
+  it('can make an unauthenticated request', async () => {
+    service.isAuthenticated = false;
+    // Request should be rejected with an error if it's not authenticated.
+    assert.rejects(async () => {
+      await request(service, markdown)
+    }, {
+      name: 'Error', 
+      message: 'Renderer service could not respond to request: '
+    });
+  });
+
+  it('can make an authenticated request with an invalid url', async () => {
+    service.isAuthenticated = true;
+    // Request will be rejected if it's authenticated but given an invalid url.
+    assert.rejects(async () => {
```

Comment:
I don't believe I've seen this structure actually used. Dong a bit of [background research](https://stackoverflow.com/questions/54247064/mocha-node-how-to-use-assert-rejects), I wonder if this will work as expected. Have you confirmed?

For readability, I tend to prefer simpler assert methods, so in this case would probably use something like:

```
try {
  const response = await request(service, markdown);
  assert.ok(response.body.length > 0, 'Metadata server sent empty value')
} catch (e) {
  assert.fail('Metadata server could not respond to request: ')
}
```

Reply:
I can now confirm that it does **not** work as expected. I also now understand *why* it does not work as expected.

I've updated the test with a structure similar to yours, that I have tested and confirmed that it works.